### ISSUE TITLE
Navigation Block: Flip submenu indicator arrow icon on click/hover

### ISSUE THREAD

**Issue by jacobcassidy** [WordPress/gutenberg#35636]
## What problem does this address?

Currently, when you open a submenu, the downward arrow icon does not change. Ideally, it would flip into an upward arrow when the submenu is open.

## What is your proposed solution?

Add a CSS transform to flip the SVG arrow icon. For example, using the following CSS for on-click submenus:

```
.wp-block-navigation-submenu__toggle[aria-expanded=true] .wp-block-navigation__submenu-icon svg {
    transform: scaleY(-1);
}
```
![closed-open_submenu](https://user-images.githubusercontent.com/1895973/137321683-d303cb80-64db-4fc5-96b8-3ea837d67c12.png)

**Comment by karmatosed** [comment]
I actually think this would be a nice addon because it does look a bit weird to still have it showing down when it is open. That said is is a minimal menu not a full designed one. I think getting a PR to try would be nice and take from there.

**Comment by shrivastavanolo** [comment]
Hey @jacobcassidy please let me know if this is still relevant. I can work on this. 

cc: @talldan

**Comment by t-hamano** [comment]
Reopening this issue to fix the problem introduced by #70307. See https://github.com/WordPress/gutenberg/pull/70307#issuecomment-2943569282

**Comment by t-hamano** [comment]
There are two points to fix this problem:

- The thing to flip should be `.wp-block-navigation__submenu-icon`, not `.wp-block-navigation-submenu__toggle`.
- We will likely need to flip on the X axis as well as the Y axis.

**Comment by shrivastavanolo** [comment]
@t-hamano Yes you are right about that. The second point can be fixed by using rotate(180deg) instead of scaleY(-1) so that it rotates the svg by 180 degrees instead of flipping along the Y axis. I'll be making a PR to fix this shortly. Thank you for your help!

**Comment by shrivastavanolo** [comment]
Closed #70338 as duplicate PR

**Comment by t-hamano** [comment]
@karmatosed @fabiankaegy @shrivastavanolo I'd like to revert #70307 so that the issue doesn't ship in the upcoming Gutenberg 21.1.

A fix for this issue is in progress in #70334, but let's test carefully and consider various scenarios.

**Comment by shrivastavanolo** [comment]
Hey @fabiankaegy Is it okay if I create a fresh PR for this with the changes mentioned in https://github.com/WordPress/gutenberg/pull/70334#pullrequestreview-2900244815 as I really do believe this change will help improve the UX. No worries if the change is not required or if you plan to work on https://github.com/WordPress/gutenberg/pull/70334 though.

cc: @t-hamano
